driver-async/src/main/com/mongodb/async/client/gridfs/GridFSIndexCheckImpl.java

```diff
@@ -28,17 +29,17 @@
 
 final class GridFSIndexCheckImpl implements GridFSIndexCheck {
     private static final Document PROJECTION = new Document("_id", 1);
-    private final MongoCollection<Document> filesCollection;
+    private final MongoCollection<GridFSFile> filesCollection;
     private final MongoCollection<Document> chunksCollection;
 
-    GridFSIndexCheckImpl(final MongoCollection<Document> filesCollection, final MongoCollection<Document> chunksCollection) {
+    GridFSIndexCheckImpl(final MongoCollection<GridFSFile> filesCollection, final MongoCollection<Document> chunksCollection) {
         this.filesCollection = notNull("files collection", filesCollection);
         this.chunksCollection = notNull("chunks collection", chunksCollection);
     }
 
     @Override
     public void checkAndCreateIndex(final SingleResultCallback<Void> callback) {
-        filesCollection.withReadPreference(primary()).find().projection(PROJECTION).first(
+        filesCollection.withDocumentClass(Document.class).withReadPreference(primary()).find().projection(PROJECTION).first(
```

Comment:
Because we only care if there are documents in there, not if they are valid GridFS documents